roles/core/pxe_stack/templates/nodes_parameters.yml.j2

```
@@ -1,8 +1,19 @@
 #jinja2: lstrip_blocks: "True"
-{% import "macros/network.j2" as macro_network with context %}
 #### Blue Banquise file ####
 ## {{ ansible_managed }}
 
+{% macro node_main_network(macro_network_interfaces,macro_current_iceberg_network) %}
+{{ macro_network_interfaces | selectattr('network','match','^'+macro_current_iceberg_network+'-[a-zA-Z0-9]+') | map(attribute='network') | list | first }}
+{% endmacro %}
+
+{% macro node_main_network_interface(macro_network_interfaces,macro_current_iceberg_network) %}
+{{ macro_network_interfaces | selectattr('network','match','^'+macro_current_iceberg_network+'-[a-zA-Z0-9]+') | map(attribute='interface') | list | first }}
+{% endmacro %}
+
+{% macro node_main_network_ip4(macro_network_interfaces,macro_current_iceberg_network) %}
+{{ macro_network_interfaces | selectattr('network','match','^'+macro_current_iceberg_network+'-[a-zA-Z0-9]+') | map(attribute='ip4') | list | first }}
+{% endmacro %}
+
```

Wondering if we could factorize here to get the main network element from network_interfaces with a single macro, then map the attributes later:
```suggestion
{% macro node_main_network(macro_network_interfaces,macro_current_iceberg_network) %}
{{ macro_network_interfaces | selectattr('network','match','^'+macro_current_iceberg_network+'-[a-zA-Z0-9]+') | first }}
{% endmacro %}
```
(see below for map())

---

We can simply pass the attribute as argument.

---

With the attribute as an argument, it will be 3 calls to the macro (one per attribute).

---

👍